Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -459,7 +459,7 @@ void TextureCacheBase::SerializeTexture(AbstractTexture* tex, const TextureConfi
     // needing to allocate/free an extra buffer.
     u8* texture_data = p.DoExternal(total_size);
 
-    if (p.GetMode() == PointerWrap::MODE_MEASURE)
+    if (!skip_readback && p.GetMode() == PointerWrap::MODE_MEASURE)
```

Comment:
Did `p`'s mode potentially change?  Kinda surprised you can't just do `if (skip_readback)`?

Reply:
Theoretically, the mode can change to measure (see e.g. [this](https://github.com/dolphin-emu/dolphin/blob/2e01dc0c82b5f73215e3425ab45ff6c7ee50b389/Source/Core/Core/State.cpp#L152-L173)).  Not sure if that's relevant in this case though.

Reply:
It is relevant, more specifically because of this: https://github.com/dolphin-emu/dolphin/blob/2e01dc0c82b5f73215e3425ab45ff6c7ee50b389/Source/Core/Common/ChunkFile.h#L219-L223

Reply:
Thanks, sorry I didn't look at `DoExternal` but wondered if that was the case.  I've never looked at Dolphin's serialization code  Not to derail this PR but that code seems surprising.  Feels like it should return `nullptr`...

Anyway, appreciate the explanation.  Just thought I'd take the opportunity to learn!